tests/unit/test_vault.py

```diff
@@ -95,56 +41,64 @@ def test_given_n_unseal_keys_when_unseal_then_unseal_called_n_times(
             [call(unseal_key) for unseal_key in unseal_keys]
         )
 
-    @patch("vault.Vault.is_ready")
-    def test_given_is_ready_when_wait_to_be_ready_then_returns(self, patch_is_ready):
-        patch_is_ready.return_value = True
-
+    @patch("hvac.api.system_backend.health.Health.read_health_status")
+    def test_given_connection_error_when_is_api_available_then_return_false(
+        self, patch_health_status
+    ):
+        patch_health_status.side_effect = requests.exceptions.ConnectionError()
         vault = Vault(url="http://whatever-url")
 
-        vault.wait_to_be_ready(timeout=1)
+        self.assertFalse(vault.is_api_available())
 
-    @patch("vault.Vault.is_ready")
-    @patch("vault.sleep", new=Mock)
-    def test_given_is_not_ready_when_wait_to_be_ready_then_timeout_error(self, patch_is_ready):
-        patch_is_ready.return_value = False
+    @patch("hvac.api.system_backend.health.Health.read_health_status")
+    def test_given_api_returns_when_is_api_available_then_return_true(self, patch_health_status):
+        patch_health_status.return_value = requests.Response()
         vault = Vault(url="http://whatever-url")
 
-        with self.assertRaises(TimeoutError):
-            vault.wait_to_be_ready(timeout=1)
+        self.assertTrue(vault.is_api_available())
 
-    @patch("vault.Vault.is_api_available")
-    def test_given_api_available_when_wait_for_api_available_then_returns(
-        self, patch_is_api_available
+    @patch("hvac.api.system_backend.raft.Raft.read_raft_config")
+    def test_given_node_in_peer_list_when_is_node_in_raft_peers_then_returns_true(
+        self, patch_health_status
     ):
-        patch_is_api_available.return_value = True
+        node_id = "whatever node id"
         vault = Vault(url="http://whatever-url")
+        patch_health_status.return_value = {
+            "data": {"config": {"servers": [{"node_id": node_id}]}}
+        }
 
-        vault.wait_for_api_available(timeout=1)
+        self.assertTrue(vault.is_node_in_raft_peers(node_id=node_id))
 
-    @patch("vault.sleep", new=Mock)
-    @patch("vault.Vault.is_api_available")
-    def test_given_api_not_available_when_wait_for_api_available_then_timeouterror(
-        self,
-        patch_is_api_available,
+    @patch("hvac.api.system_backend.raft.Raft.read_raft_config")
+    def test_given_node_not_in_peer_list_when_is_node_in_raft_peers_then_returns_false(
+        self, patch_health_status
     ):
-        patch_is_api_available.return_value = False
+        node_id = "whatever node id"
         vault = Vault(url="http://whatever-url")
```

Comment:
Just a minor thing, Vault URL could be stored in a variable. "http://whatever-url" is set multiple times.

Reply:
I think it's ok that it's initialised multiple times, it makes each test readable. I'd rather optimise for readability then efficiency